_posts/2017-11-09-totw-135.md

```diff
@@ -0,0 +1,83 @@
+---
+title: "Tip of the Week #135: Test the Contract, not the Implementation"
+layout: tips
+sidenav: side-nav-tips.html
+published: draft
+permalink: tips/135
+type: markdown
+---
+
+# Tip of the Week #135: Test the Contract, not the Implementation
+
+Originally posted as totw/135 on ORIGINAL DATE
+
+By James Dennett (jdennett@google.com)
+
+"If you have one true friend you have more than your share." — Thomas Fuller
+
+C++ has a somewhat elaborate access control mechanism using `public`,
+`protected`, `private`, and `friend`. Test code has its own rules of etiquette
+for using those facilities, and GoogleTest augments them with its own
+`FRIEND_TEST` macro. Use of `FRIEND_TEST` should be a last resort, not a
+preferred choice.
+
```

Comment:
macro comment. Your (and James`) call, but this tip is "texty" and I'd prefer it to have a few more headings to separate out the logical thread here. That thread seems to be:

"Test the Contract"
"Augmenting the Public API for Tests"
"Using Peers to Avoid Exposing Implementation"

Reply:
I suppose. PTAL.

Reply:
LGTM